moto/iot/models.py

```diff
@@ -0,0 +1,364 @@
+        self.arn = 'arn:aws:iot:%s:1:thing/%s' % (self.region_name, thing_name)
+        self.version = 1
+        # TODO: we need to handle 'version'?
+
+        # for iot-data
+        self.thing_shadow = None
+
+    def to_dict(self, include_default_client_id=False):
+        obj = {
+            'thingName': self.thing_name,
+            'attributes': self.attributes,
+            'version': self.version
+        }
+        if self.thing_type:
+            obj['thingTypeName'] = self.thing_type.thing_type_name
+        if include_default_client_id:
+            obj['defaultClientId'] = self.thing_name
+        return obj
+
+
+class FakeThingType(BaseModel):
+    def __init__(self, thing_type_name, thing_type_properties, region_name):
+        self.region_name = region_name
+        self.thing_type_name = thing_type_name
+        self.thing_type_properties = thing_type_properties
+        t = time.time()
+        self.metadata = {
+            'deprecated': False,
+            'creationData': int(t * 1000) / 1000.0
+        }
+        self.arn = 'arn:aws:iot:%s:1:thingtype/%s' % (self.region_name, thing_type_name)
+
+    def to_dict(self):
+        return {
+            'thingTypeName': self.thing_type_name,
+            'thingTypeProperties': self.thing_type_properties,
+            'thingTypeMetadata': self.metadata
+        }
+
+
+class FakeCertificate(BaseModel):
+    def __init__(self, certificate_pem, status, region_name):
+        m = hashlib.sha256()
+        m.update(str(uuid.uuid4()).encode('utf-8'))
+        self.certificate_id = m.hexdigest()
+        self.arn = 'arn:aws:iot:%s:1:cert/%s' % (region_name, self.certificate_id)
+        self.certificate_pem = certificate_pem
+        self.status = status
+
+        # TODO: must adjust
+        self.owner = '1'
+        self.transfer_data = {}
+        self.creation_date = time.time()
+        self.last_modified_date = self.creation_date
+        self.ca_certificate_id = None
+
+    def to_dict(self):
+        return {
+            'certificateArn': self.arn,
+            'certificateId': self.certificate_id,
+            'status': self.status,
+            'creationDate': self.creation_date
+        }
+
+    def to_description_dict(self):
+        """
+        You might need keys below in some situation
+          - caCertificateId
+          - previousOwnedBy
+        """
+        return {
+            'certificateArn': self.arn,
+            'certificateId': self.certificate_id,
+            'status': self.status,
+            'certificatePem': self.certificate_pem,
+            'ownedBy': self.owner,
+            'creationDate': self.creation_date,
+            'lastModifiedDate': self.last_modified_date,
+            'transferData': self.transfer_data
+        }
+
+
+class FakePolicy(BaseModel):
+    def __init__(self, name, document, region_name):
+        self.name = name
+        self.document = document
+        self.arn = 'arn:aws:iot:%s:1:policy/%s' % (region_name, name)
```

Comment:
Just a thought, we have a fair amount of utils.py files with functions that create arns, couldn't we generalise this with 'arn:aws:iot:%s:1:%s/%s'.

Haven't looked at iot, but is the 1 in the arn hardcoded or does it increment at all.

Reply:
Yes, we can try to generalize function of generating arn.
But as long as I see `utils.py`s in service directories, the structure of arn differs by service.
And account id will be same in many cases, so setting it to `1` here seems to make sense.

We should make generalized flow of defining FakeObjects though.

Reply:
Ahh fair enough 

> We should make generalized flow of defining FakeObjects though.

Yep sounds good.